### DISCUSSION TITLE
API to download backups

### DISCUSSION THREAD

**Discussion by timkelty** [craftcms/cloud#18] [Feature Requests & Enhancements]
If you want your own copy of a database dump, the only option is to download it from Console.

An API would be great, so it could be automated.

**Comment by svale** [comment]
From a redundancy perspective, this would be a very welcome feature if it could used to implement a remote/secondary backup storage location.

**Reply by jamesmacwhite** [reply]
Currently we connect to the RDS instance and run mysqlbackup on scheduled points and upload to a separate S3 bucket via the AWS CLI. A direct API would be very welcome however. We still think that having an additional backup location outside of Craft Cloud is a good thing. We have observed occasional silent backup failures from the automated backups.

The other benefit is we can gzip the s3 backups, Craft Cloud backups currently aren't.

**Reply by svale** [reply]
Ah, thanks, that is a good tip! I assume you are running that as an automated job - can I ask on which plattform you've set that up?

**Reply by jamesmacwhite** [reply]
Yes, we are. We have a twice a day backup policy for the production environment and every 24 hours for non-live environemtns, given Craft Cloud does not do automatic backups for non-production environments at all.

We use Buddy.works, but anything that can run a container with the mysqlbackup binary available will do. This could be an Ubuntu container with mysqlbackup installed or the MySQL image that provides mysqlbackup.

The backing up without Craft guidance provides the mysqlbackup command to use: https://craftcms.com/knowledge-base/cloud-databases#backing-up-without-craft, that doesn't cause major issues if run during peak times. The only consideration is the version of mysqlbackup provided in a container may not directly match Craft Cloud. We note a slight variance e.g. version 8.0.39 on Craft Cloud compared to 8.0.40. This should mostly be fine providing the difference isn't too far.

We test a combination of Craft Cloud and our own S3 backups for local environments and syncing non-live environments to validate backups.

**Reply by svale** [reply]
Nice! And thanks for the detailed write up 🙌

**Comment by andrewfairlie** [comment]
> If you want your own copy of a database dump, the only option is to download it from Console.
> 
> An API would be great, so it could be automated.

It would also be great if there was a corresponding Craft CLI to download the latest backup (or even live DB) for your local dev environment `ddev craft cloud/db pull` or something.
